service/create/service.go

```diff
@@ -492,10 +519,21 @@ func (s *Service) runMachine(input runMachineInput) error {
 			},
 		},
 	}); err != nil {
-		return microerror.MaskAny(err)
+		return nil, microerror.MaskAny(err)
 	}
 
 	s.logger.Log("info", fmt.Sprintf("instance '%s' tagged", input.name))
+	instance := reservation.Instances[0]
 
-	return nil
+	return instance, nil
```

Comment:
We would have to call something like [DescribeInstances](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.DescribeInstances), filtering the query using some combination of tags (for example, only masters), Regions, AZs and VPC IDs, since `DescribeInstances` by default returns all instances owned by an account, which is definitely too broad.

With that, we could get the Instance ID of the master (in this stage, there's only one), and then go on with the Elastic IP.